test/main_dialog_spec.rb

```diff
@@ -0,0 +1,174 @@
+require_relative "spec_helper.rb"
+require "update-alternatives/UI/main_dialog"
+
+describe UpdateAlternatives::MainDialog do
+  def mock_ui_events(*events)
+    allow(Yast::UI).to receive(:UserInput).and_return(*events)
+  end
+
+  subject(:dialog) { UpdateAlternatives::MainDialog.new }
+
+  before do
+    allow(Yast::UI).to receive(:OpenDialog).and_return(true)
+    allow(Yast::UI).to receive(:CloseDialog).and_return(true)
+    allow(UpdateAlternatives::Alternative).to receive(:all)
+      .and_return(main_dialog_alternatives_list)
+  end
+
+  def expect_update_table_with(expected_items)
+    expect(Yast::UI).to receive(:ChangeWidget) do |widgetId, option, itemsList|
+      expect(widgetId).to eq(:alternatives_table)
+      expect(option).to eq(:Items)
+      expect(itemsList.map(&:params)).to eq(expected_items)
+    end
+  end
+
+  describe "#run" do
+    it "ignores EmptyAlternative ojects" do
+      dialog.instance_variable_get(:@alternatives_list).each do |alternative|
+        expect(alternative).not_to be_an(UpdateAlternatives::EmptyAlternative)
+      end
+    end
+  end
+
+  describe "#multi_choice_only_handler" do
+    before do
+      mock_ui_events(:multi_choice_only, :cancel)
+    end
+
+    context "if multi_choice_only filter is enabled" do
+      it "update the table of alternative excluding the alternatives with only one choice" do
+        allow(Yast::UI).to receive(:QueryWidget).with(:multi_choice_only, :Value).and_return(true)
```

Comment:
Nitpicking (feel free to ignore): since you are using `context` here, the `allow` part would probably correspond to a `before` in the `context`. But that's probably overkill anyways.